lab_1_keywords_tfidf/main.py

```diff
@@ -157,6 +276,34 @@ def calculate_expected_frequency(
         dict[str, float] | None: Dictionary with expected frequencies.
         In case of corrupt input arguments, None is returned.
     """
+    if not isinstance(doc_freqs, dict) or not isinstance(corpus_freqs, dict):
+        return None
+
+    total_doc_tokens = 0
+    for token, count in doc_freqs.items():
+        if not isinstance(token, str) or not isinstance(count, int) or count < 0:
+            return None
+        total_doc_tokens += count
+
+    total_corpus_tokens = 0
+    for token, count in corpus_freqs.items():
+        if not isinstance(token, str) or not isinstance(count, int) or count < 0:
+            return None
+        total_corpus_tokens += count
+
+    if total_doc_tokens == 0:
+        return {token: 0.0 for token in doc_freqs}
+
+    if total_corpus_tokens == 0:
+        return None 
+
+    expected_frequencies = {}
+    for token, doc_count in doc_freqs.items():
+        corpus_count = corpus_freqs.get(token, 0) 
+        expected_freq = (total_doc_tokens / total_corpus_tokens) * corpus_count
+        expected_frequencies[token] = expected_freq
+
+    return expected_frequencies
```

Comment:
Are you sure, that you correctly calculate expected frequency according to documentation?

Reply:
Why? It is not hard(